### PR TITLE
Update clang-format rules to fit ROS 2 style guide

### PR DESCRIPTION
**Pull request by james-rms**
Directly uses the ROS 2 clang-format rules from [here](https://github.com/ament/ament_lint/blob/foxy/ament_clang_format/ament_clang_format/configuration/.clang-format). This also brings rosbag2_storage_mcap into compliance with the ROS 2 style guide.

Some notable changes:
(https://google.github.io/styleguide/cppguide.html#Preprocessor_Directives).
* The ROS 2 style guide specifies [pointer alignment](https://docs.ros.org/en/foxy/The-ROS2-Project/Contributing/Code-Style-Language-Versions.html#pointer-syntax-alignment) differently from our previous rule.
* Line breaks before braces have different rules in the [ROS 2 style guide](https://docs.ros.org/en/foxy/The-ROS2-Project/Contributing/Code-Style-Language-Versions.html#open-versus-cuddled-braces).

## Testing
- [x] `ament_clang_format .` returns no errors from repo root
- [x] `ament_cpplint .` returns no errors from repo root

### CODE REVIEW

File: rosbag2_storage_mcap/include/rosbag2_storage_mcap/visibility_control.hpp [hunk 1]
```diff
@@ -24,26 +24,26 @@
...
-  #else
-    #define ROSBAG2_STORAGE_MCAP_EXPORT __declspec(dllexport)
-    #define ROSBAG2_STORAGE_MCAP_IMPORT __declspec(dllimport)
-  #endif
-  #ifdef ROSBAG2_STORAGE_MCAP_BUILDING_DLL
-    #define ROSBAG2_STORAGE_MCAP_PUBLIC ROSBAG2_STORAGE_MCAP_EXPORT
-  #else
-    #define ROSBAG2_STORAGE_MCAP_PUBLIC ROSBAG2_STORAGE_MCAP_IMPORT
-  #endif
+#ifdef __GNUC__
```
Comment (emersonknapp) [inline rosbag2_storage_mcap/include/rosbag2_storage_mcap/visibility_control.hpp]: This one's a bit much - to me. That googlec++ style seemed to mean that you don't indent _with other code_ but it didn't seem to have anything to say about full-preprocessor blocks and indentation. This file is indented in other packages, but those generally aren't using `clang-format` yet

**Review by jhurliman — APPROVED** [review]

File: rosbag2_storage_mcap/include/rosbag2_storage_mcap/message_definition_cache.hpp [hunk 2]
```diff
@@ -84,15 +89,15 @@ class MessageDefinitionCache final {
...
 
   std::unordered_map<DefinitionIdentifier, MessageSpec, DefinitionIdentifierHash>
     msg_specs_by_definition_identifier_;
 };
 
 ROSBAG2_STORAGE_MCAP_PUBLIC
-std::set<std::string> parse_dependencies(Format format, const std::string& text,
-                                         const std::string& package_context);
+std::set<std::string> parse_dependencies(
+  Format format, const std::string & text, const std::string & package_context);
```
Comment (MichaelOrlov) [inline rosbag2_storage_mcap/include/rosbag2_storage_mcap/message_definition_cache.hpp]: @james-rms I am curious if it's possible to preserve original formatting with indentation on second line with argument?

**Review by emersonknapp — APPROVED** [review]

File: rosbag2_storage_mcap/include/rosbag2_storage_mcap/message_definition_cache.hpp (hunk 2 above)
Comment (james-rms) [reply rosbag2_storage_mcap/include/rosbag2_storage_mcap/message_definition_cache.hpp]: We can if we remove the `AlignAfterOpenBracket: AlwaysBreak` line from `.clang-format`. However I'm not sure if i'm breaking some rule by doing this that's an explicit directive in `ament_clang_format`'s configuration, so i feel like it's there for a reason.

Comment (james-rms) [reply rosbag2_storage_mcap/include/rosbag2_storage_mcap/message_definition_cache.hpp]: https://google.github.io/styleguide/cppguide.html#Function_Declarations_and_Definitions it looks like it should be fine either way, so i'll remove that line.

**Comment by james-rms** [comment]
CI is failing on `rolling` because `update_metadata` is not implemented, which I will do in a separate PR.

**Pull request merged**